pkg/secrets/fetch_secret.go

```diff
@@ -157,3 +197,41 @@ func fetchSecret(secretsHandle []string, origin string) (map[string]string, erro
 	}
 	return res, nil
 }
+
+func fileHashSHA256(f *os.File) (string, error) {
+	stat, err := f.Stat()
+	if err != nil {
+		return "", err
+	}
+
+	h := sha256.New()
+
+	fileSize := stat.Size()
+	if fileSize == 0 {
+		return formatHash(h), nil
+	}
+
+	var bufSize int64 = 102400
+	if fileSize < bufSize {
+		bufSize = fileSize
+	}
+
+	buffer := make([]byte, bufSize)
+
+	for {
+		read, err := f.Read(buffer)
```

Comment:
It would be best if you implemented a limiter to ensure that the agent can't be blocked at the bootstrap time while it tries to hash the target file.

Also, consider migrating to `io.CopyN` to read and stream directly.

Reply:
There is a timeout in place for execution of secret backends [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/secrets/fetch_secret.go#L47-L48). It will include the hashing part. Default is 5s. My fear of using `CopyN` is in making it more difficult to disambiguate the source of errors.

Reply:
I was looking more for a memory limiter than a time limiter. Currently, if the user set `/dev/zero` as the target script, the sha256 computation will block due to the infinite nature of `/dev/zero` (aside from its type check). This could be used to prevent an agent from starting, and in the worst case, it could generate a heavy workload on the computer itself. You could add an acceptable script file size check above and ensure the reader is truncated/ended by an `io.LimitReader`.